src/engine/sync/synccontrol.cpp

```diff
@@ -411,43 +410,58 @@ void SyncControl::slotSyncEnabledChangeRequest(double enabled) {
 }
 
 void SyncControl::setLocalBpm(double local_bpm) {
-    if (getSyncMode() == SYNC_NONE) {
+    if (local_bpm == m_prevLocalBpm.getValue()) {
         return;
     }
-    if (local_bpm == m_prevLocalBpm) {
-        return;
-    }
-    if (local_bpm == 0 && m_pPlayButton->toBool()) {
-        // If the local bpm is suddenly zero and sync was active and we are playing,
-        // stick with the previous localbpm.
-        // I think this can only happen if the beatgrid is reset.
-        qWarning() << getGroup() << "Sync is already enabled on track with empty or zero bpm";
+    m_prevLocalBpm.setValue(local_bpm);
+
+    SyncMode syncMode = getSyncMode();
+    if (syncMode <= SYNC_NONE) {
```

Comment:
!isSynchronized()?

Reply:
syncMode is tested below explicit for the other values as well. 
IMHO it is better to read if keep the comparison here as well.